uberenv.py

```diff
@@ -493,32 +521,51 @@ def install(self):
         # note: this assumes package extends python when +python
         # this may fail general cases
         if self.opts["install"] and "+python" in full_spec:
-            activate_cmd = "spack/bin/spack activate " + self.pkg_name
+            activate_cmd = "spack/bin/spack activate " + self.pkg_name + self.opts["spec"]
```

Comment:
Since ``spack`` commands now need additional context/environments, would it make sense to go through a utility function, e.g. ``create_spack_command()`` that deals with the path to ``spack`` and adding the appropriate environment? This could help improve consistency.

This could definitely be handled in a separate PR.

Reply:
This comment might be more appropriate in the context of https://github.com/LLNL/uberenv/pull/18